### PR TITLE
tree-wide: drop several remaining license headers

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/utf8.c [hunk 1]
```diff
@@ -1,26 +1,10 @@
...
-
-/* gutf8.c - Operations on UTF-8 strings.
+/* Parts of this file are based on the GLIB utf8 validation functions. The original copyright follows.
  *
+ * gutf8.c - Operations on UTF-8 strings.
  * Copyright (C) 1999 Tom Tromey
  * Copyright (C) 2000 Red Hat, Inc.
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Library General Public
```
Comment (bluca) [inline src/basic/utf8.c:12]: Uuuuh this file is lgpl but this mentions gpl... this is not good, we need to audit it, can you please leave it for now

Comment (bluca) [reply src/basic/utf8.c:12]: Ah no sorry I misread, was searching for 'lesser', guess it's the older wording in 2.0 vs 2.1

Comment (bluca) [reply src/basic/utf8.c:12]: However given it's a different license than the file, although compatible, we should keep a reference

Comment (keszybz) [reply src/basic/utf8.c:12]: Hmm, I think we should go in the other direction. Still drop the boilerplate text, but change the  SPDX-License-Identifier to LGPL-2.0-or-later.

Comment (yuwata) [reply src/basic/utf8.c:12]: @keszybz You mean GPL, rather than LGPL?

Comment (bluca) [reply src/basic/utf8.c:12]: No this file is LGPL-2.0-or-later given there's a code snippet  under that license

Comment (yuwata) [reply src/basic/utf8.c:12]: Done. Also the README file is updated.

**Pull request merged**